sycl/source/detail/platform_impl.cpp

```diff
@@ -28,10 +28,19 @@ using PlatformImplPtr = std::shared_ptr<platform_impl>;
 
 PlatformImplPtr platform_impl::getHostPlatformImpl() {
   static PlatformImplPtr HostImpl = std::make_shared<platform_impl>();
-
   return HostImpl;
 }
 
+std::shared_ptr<device_impl> platform_impl::getOrMakeHostDeviceImpl() {
+  PlatformImplPtr HostImpl = getHostPlatformImpl();
+  if (HostImpl->MDeviceCache.size() == 0) {
```

Comment:
It seems strange that there are two device caches now. Suggest merging device and platform caches into one map with a platform as a key and a vector of devices as a value. If this was done the device cache in the `platform_impl` can be removed.

Reply:
I merged two separate caches for `device` and `platform`.